### PR TITLE
Add group summary APIs

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/group_server.py [hunk 1]
```diff
@@ -64,6 +71,531 @@ def get_rooms_in_group(self, group_id, include_private=False):
...
+                },
+                retcol="group_id",
+                allow_none=True,
+            )
+            if not role_exists:
+                # If not, add it with an order larger than all others
+                txn.execute("""
+                    INSERT INTO group_summary_roles
+                    (group_id, role_id, role_order)
+                    SELECT ?, ?, COALESCE(MAX(role_order), 1)
```
Comment (NegativeMjark) [inline synapse/storage/group_server.py]: Will this be larger than all the others, or the same as the largest?

File: synapse/storage/group_server.py [hunk 2]
```diff
@@ -15,11 +15,18 @@
...
+from synapse.api.errors import SynapseError
+
 from ._base import SQLBaseStore
 
 import ujson as json
 
 
+# The category ID for the "default" category. We don't store as null in the
+# database to avoid the fun of null != null
+_DEFAULT_CATEGORY_ID = "default"
```
Comment (NegativeMjark) [inline synapse/storage/group_server.py]: Maybe use the empty string as the default ID here?

File: synapse/federation/transport/server.py [hunk 3]
```diff
@@ -750,6 +766,220 @@ def on_POST(self, origin, content, query, group_id, user_id):
...
+        - /groups/:group/summary/categories/:category/rooms/:room_id
+    """
+    PATH = (
+        "/groups/(?P<group_id>[^/]*)/summary"
+        "(/categories/(?P<category_id>[^/]+))?"
+        "/rooms/(?P<room_id>[^/]*)$"
+    )
+
+    @defer.inlineCallbacks
+    def on_POST(self, origin, content, query, group_id, category_id, room_id):
```
Comment (NegativeMjark) [inline synapse/federation/transport/server.py]: Maybe check that the category is not the default ID.

Comment (erikjohnston) [inline synapse/federation/transport/server.py]: (done)

**Pull request merged**